src/controllers/project.js

```diff
@@ -46,6 +48,105 @@ const supportProject = async (req, res, next) => {
   return next();
 };
 
+const createReview = async (req, res, next) => {
+  const { id } = req.params;
+  const { userID, rating, content } = req.body;
+
+  try {
+    const project = await Project.findOne({ _id: id });
+
+    if (!project) return res.status(404).json({ message: 'Project not found' });
+
+    if (
+      project.reviews.find(
+        (review) => ObjectId(review.user).toString() === userID
+      )
+    )
+      return res.json({ message: 'You already reviewed this project' });
+
+    const review = {
+      user: userID,
+      rating,
+      content,
+    };
+
+    project.reviews.push(review);
+
+    await project.save();
+
+    res.json({
+      message: 'Review created successfully',
+      reviews: project.reviews,
+    });
+  } catch (error) {
+    res.status(422).json({ message: 'Unable to create review' });
+  }
+  return next();
+};
+
+const isNotEmpty = (field) => !!field;
+
+const updateReview = async (req, res, next) => {
+  const { id, reviewId } = req.params;
+  const { rating, content } = req.body;
+
+  try {
+    const project = await Project.findById(id);
+
+    if (!project) return res.status(404).json({ message: 'Project not found' });
+
+    const review = project.reviews.find(
+      // eslint-disable-next-line no-underscore-dangle
+      (single) => ObjectId(single._id).toString() === reviewId
+    );
+
+    if (!review) return res.status(404).json({ message: 'Review not found' });
+
+    review.rating = isNotEmpty(rating) ? rating : review.rating;
```

Comment:
I mean I still don't see the idea of this function isNotEmpty() why cant you just write 
```
rating || review.rating
```
I think this will give the same effect with less code and less confusion

Reply:
@felmez this is still not fixed

Reply:
its not a problem or bug ya, its just preferred thing up to a person, why i should change it i mean where is the standard for this ??